linux/library/src/flutter_embedder.cc

```diff
@@ -11,18 +11,79 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-#include "include/flutter_embedder.h"
+#include <flutter/flutter_embedder.h>
 
-#include <GL/gl.h>
+#include <X11/Xlib.h>
 #include <assert.h>
+#include <gtk/gtk.h>
+#include <json/json.h>
+
 #include <chrono>
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <string>
 
-#include "include/embedder.h"
+#include <flutter/embedder_extensions.h>
+#include <flutter/file_chooser_plugin.h>
+#include <flutter/plugin_handler.h>
 
 static_assert(FLUTTER_ENGINE_VERSION == 1, "");
 
+// Struct for storing state within an instance of the GLFW Window.
+struct FlutterEmbedderState {
+  FlutterEngine engine;
+  std::unique_ptr<flutter::PluginHandler> plugin_handler;
+};
+typedef FlutterEmbedderState *FlutterEmbedderStatePtr;
```

Comment:
Why the typedef? The typedef version is longer, and makes it less clear what kind of pointer it is (raw vs. managed)

Reply:
Initially just laziness.  Fixed with the `GetSavedEmbedderState` function.